### ISSUE TITLE
Lottery isn't assigning lunch blocks

### ISSUE THREAD

**Issue by benjaminjkraft** [learning-unlimited/ESP-Website#1107]
(Modulo some unrelated bugs,) the lottery makes sure that each student has a block free for lunch.  However, it doesn't actually assign them to lunches.  It's easier for everyone if they already get assigned to lunches, so the lottery should do this.  Ideally, it should assign students who have both lunch blocks free so as to equalize the number of students with each lunch block.

**Comment by willgearty** [comment]
See `esp/useful_scripts/add_lunches.py`

**Comment by SurbhiAgarwal1** [comment]
Hi @willgarty!

I noticed this issue has been open since 2014 and is assigned to you. I'm interested in working on fixing the lottery lunch block assignment logic.

Is this still an active issue that needs to be fixed? If so, would you be open to me working on a solution? I'd be happy to investigate the lottery system and submit a PR if you can provide some guidance on:

1. Is this bug still occurring in the current codebase?
2. Are there any specific test cases or scenarios I should focus on?
3. Any particular areas of the lottery code I should look at first?

I have experience with the ESP codebase (currently have PRs #4267 and #4327 under review) and would love to help tackle this if you're not actively working on it.

Thanks!

**Comment by willgearty** [comment]
@SurbhiAgarwal1 Thanks for your interest in contributing to our project! I've assigned you to this issue. Please make sure to check out our [instructions for getting started](https://github.com/learning-unlimited/ESP-Website/wiki#getting-started) and follow our [contribution rules](https://github.com/learning-unlimited/ESP-Website/wiki#rules). Let me know if you have any questions.

**Comment by willgearty** [comment]
Yes, this remains an issue. The specific scenario is if you have multiple lunches in a single day (check out the Lunch Constraints Module), you want to assign students to one of those lunches as part of their schedule during the class lottery assignment (you don't want students to have a schedule full of classes and not have time for lunch, as much as THEY may want that). You'll want to add this functionality to the Lottery Controller: https://github.com/learning-unlimited/ESP-Website/blob/main/esp/esp/program/controllers/lottery.py
